src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.ts

```diff
@@ -1,61 +1,86 @@
-import { div, h, label } from 'react-hyperscript-helpers'
+import { br, div, h, label, p } from 'react-hyperscript-helpers'
 import { Link, Select } from 'src/components/common'
+import { icon } from 'src/components/icons'
 import { NumberInput } from 'src/components/input'
+import TitleBar from 'src/components/TitleBar'
 import TooltipTrigger from 'src/components/TooltipTrigger'
+import { Ajax } from 'src/libs/ajax'
 import { pdTypes } from 'src/libs/ajax/leonardo/models/disk-models'
+import Events from 'src/libs/events'
 import { useUniqueId } from 'src/libs/react-utils'
+import * as Utils from 'src/libs/utils'
 import { computeStyles } from 'src/pages/workspaces/workspace/analysis/modals/modalStyles'
+import { getCurrentMountDirectory } from 'src/pages/workspaces/workspace/analysis/utils/tool-utils'
 
+import { IComputeConfig } from '../modal-utils'
 
-interface IComputeConfig {
-  selectedPersistentDiskSize: number
-  selectedPersistentDiskType: {
-    label: string
-    displayName: string
-    regionToPricesName: string
-  }
-  masterMachineType: any
-  masterDiskSize: number
-  numberOfWorkers: number
-  numberOfPreemptibleWorkers: number
-  workerMachineType: string
-  workerDiskSize: number
-  componentGatewayEnabled: boolean
-  gpuEnabled: boolean
-  hasGpu: boolean
-  gpuType: string
-  numGpus: number
-  autopauseThreshold: number
-  computeRegion: string
-  computeZone: string
-}
 
-interface PersistentDiskProps {
-  diskExists: boolean
+export interface PersistentDiskProps {
+  persistentDiskExists: boolean
   computeConfig: IComputeConfig
   updateComputeConfig: (arg: string) => (diskType: any) => void
```

Comment:
`diskType` can be more specific than `any`

Reply:
This is currently in Dev.
Types are also out of scope for this feature